otter/util/logging_treq.py

```diff
@@ -24,16 +24,23 @@ def _log_request(treq_call, url, **kwargs):
     """
     clock = kwargs.pop('clock', reactor)
     log = kwargs.pop('log', None)
+
     if not log:
         log = default_log
     method = kwargs.get('method', treq_call.__name__)
 
-    treq_transaction = str(uuid4())
+    kwargs.setdefault('headers', {})
+    treq_transaction = kwargs['headers'].setdefault('x-otter-request-id',
+                                                    [str(uuid4())])[0]
```

Comment:
Oh. Didn't realize about the update. I am fine either way. The above code that honors user given id or ignoring it. Any one is fine as long as it is clear :) I am leaning towards ignoring user provided for now.